### PR TITLE
[civ2] python is python3, and add release infra unit tests.

### PR DESCRIPTION
**Pull request by aslonnie**
make release test infra unit tests, so that people cannot break them.

also stops running these tests on civ1.

### CODE REVIEW

File: .buildkite/cicd.rayci.yml [hunk 1]
```diff
@@ -5,3 +5,9 @@ steps:
       # TODO(aslonnie): wrap this in a script, and upload test telemetry.
       - bazel test --test_tag_filters=ci_unit //ci/ray_ci/...
     depends_on: forge
+
+  - label: "CI/CD: release test infra"
+    commands:
+      # TODO(aslonnie): wrap this in a script, and upload test telemetry.
+      - bazel test --test_tag_filters=release_unit //release/...
```
Comment (can-anyscale) [inline .buildkite/cicd.rayci.yml:12]: can we still run the ./ci/build/upload_build_info.sh here? or this pipeline doesn't have permission?

**Review by can-anyscale — APPROVED** [review]

File: .buildkite/cicd.rayci.yml (hunk 1 above)
Comment (aslonnie) [reply .buildkite/cicd.rayci.yml:12]: I think the lambda will need an update I think. I am also not sure if `upload_build_info.sh` has all its deps installed.

Comment (can-anyscale) [reply .buildkite/cicd.rayci.yml:12]: Got you, good enough for these tests!

**Pull request merged**